python/ray/data/dataset.py

```diff
@@ -4000,8 +4000,9 @@ def __repr__(self) -> str:
             schema_str = ", ".join(schema_str)
             schema_str = "{" + schema_str + "}"
         count = self._meta_count()
+        count_str = str(count) if count is not None else "?"
         return "Dataset(num_blocks={}, num_rows={}, schema={})".format(
-            self._plan.initial_num_blocks(), count, schema_str
+            self._plan.initial_num_blocks(), count_str, schema_str
```

Comment:
Nit: `str(count)` will already happen implicitly when formatting the string, so we can just replace `count` with `"?"` if it's `None`.
```python
count = self._meta_count()
if count is None:
    count = "?"
return "Dataset(num_blocks={}, num_rows={}, schema={})".format(
    self._plan.initial_num_blocks(), count, schema_str
)